services/ymax-planner/test/mocks.ts

```diff
@@ -38,9 +39,17 @@ export const mockGasEstimator = makeGasEstimator({
   fetch: mockFetchForGasEstimate,
 });
 
-export const createMockProvider = () => {
+export const createMockProvider = (
+  latestBlock = 1000,
+  eventData?: {
+    event: {};
+    block: number;
+  },
+) => {
```

Comment:
Let's not overconstrain these general mocks to injecting only single events.
```suggestion
export const createMockProvider = (
  latestBlock = 1000,
  events?: Pick<Log, 'blockNumber' | 'data' | 'topics' | 'transactionHash'>[],
): WebSocketProvider => {
```